packages/webapp/src/containers/Animals/Inventory/useAnimalOrBatchRemoval.ts

```diff
@@ -173,31 +177,52 @@ const useAnimalOrBatchRemoval = (
     return new Set([...completedTasks, ...abandonedTasks].map(({ task_id }) => task_id));
   }, [completedTasks, abandonedTasks]);
 
-  const hasFinalizedTasks = useMemo(() => {
-    if (!removalModalOpen) {
-      return false;
-    }
-
-    const selectedInventoryIdsSet = new Set(selectedInventoryIds);
-    const finalizedTaskIdsSet = getFinalizedTaskIdsSet();
+  const checkHasFinalizedTasks = async () => {
+    const animalIds: number[] = [];
+    const selectedAnimalIds: string[] = [];
+    const animalBatchIds: number[] = [];
+    const selectedBatchIds: string[] = [];
+    let result;
 
-    let inventoryFoundCount = 0;
-    for (let { id, tasks } of animalTasksWithInventoryIds) {
-      if (selectedInventoryIdsSet.has(id)) {
-        inventoryFoundCount++;
+    for (const id of selectedInventoryIds) {
+      const { kind, id: entity_id } = parseInventoryId(id);
+      if (kind === AnimalOrBatchKeys.ANIMAL) {
+        animalIds.push(entity_id);
+        selectedAnimalIds.push(id);
+      } else if (AnimalOrBatchKeys.BATCH) {
+        animalBatchIds.push(entity_id);
+        selectedBatchIds.push(id);
+      }
+    }
 
-        if (tasks.some(({ task_id }) => finalizedTaskIdsSet.has(task_id))) {
-          return true;
-        }
+    if (animalIds.length) {
+      result = await mutations['checkDeleteAnimals'].trigger(animalIds);
+      if (result.error.data === 'Cannot delete animal with completed or abandoned tasks') {
+        return true;
       }
-      // Stop iterating once all selected inventory IDs have been processed.
-      if (inventoryFoundCount === selectedInventoryIdsSet.size) {
-        return false;
+    }
+    if (animalBatchIds.length) {
+      result = await mutations['checkDeleteAnimalBatches'].trigger(animalBatchIds);
+      if (result.error.data === 'Cannot delete animal with completed or abandoned tasks') {
+        return true;
       }
     }
-  }, [removalModalOpen, completedTasks, abandonedTasks, selectedInventoryIds]);
+    return false;
```

Comment:
I'm not sure if returning false for some unknown error is expected...

Reply:
So this is an unlikely scenario -- which we could handle differently like forcing a refresh or something because it indicates a concurrency issue (or api down) since those are the only other tests. 

My thought was if there any other error -- it is okay to show the created in error option now accidentally -- because the api will now reject it if incorrect.